src/LinkedTranslation/LinkedTranslationBuilder.php

```diff
@@ -0,0 +1,67 @@
+<?php declare(strict_types=1);
+
+namespace Becklyn\Translations\LinkedTranslation;
+
+use Becklyn\HtmlBuilder\Node\SafeMarkup;
+use Becklyn\IconLoader\Registry\IconRegistry;
+use Becklyn\Rad\Route\LinkableHandlerInterface;
+use Becklyn\Rad\Route\LinkableInterface;
+use Becklyn\Translations\LinkedTranslation\Data\LinkData;
+
+class LinkedTranslationBuilder
+{
+    private $linkableHandler;
+    private $iconRegistry;
+    /** @var LinkData[] */
+    private $linkData = [];
+
+
+    public function __construct (
+        LinkableHandlerInterface $linkableHandler,
+        IconRegistry $iconRegistry
+    )
+    {
+
+        $this->linkableHandler = $linkableHandler;
+        $this->iconRegistry = $iconRegistry;
+    }
+
+
+    /**
+     * @param LinkableInterface|string|null $href
+     * @param SafeMarkup|string|null        $iconBefore
+     * @param SafeMarkup|string|null        $iconAfter
+     */
+    public function link (
+        string $beforeTag,
+        string $afterTag,
+        $href,
+        ?string $class = null,
+        $iconBefore = null,
+        $iconAfter = null
+    ) : self
+    {
+        $this->linkData[] = new LinkData($beforeTag, $afterTag, $href, $class, $iconBefore, $iconAfter);
```

Comment:
Should we check whether we already have an existing `LinkData` element with the same parameters and throw an exception? In almost all cases it's a bug if you pass in the same parameter names twice.